vector/v.info/print.c

```diff
@@ -198,30 +199,92 @@ void print_columns(struct Map_info *Map, const char *input_opt,
                 "layer <%s>:"),
               field_opt);
 
-    if ((fi = Vect_get_field2(Map, field_opt)) == NULL)
+    if ((fi = Vect_get_field2(Map, field_opt)) == NULL) {
+        Vect_close(Map);
         G_fatal_error(
             _("Database connection not defined for layer <%s> of <%s>"),
             field_opt, input_opt);
+    }
     driver = db_start_driver(fi->driver);
-    if (driver == NULL)
+    if (driver == NULL) {
+        Vect_close(Map);
         G_fatal_error(_("Unable to open driver <%s>"), fi->driver);
+    }
     db_init_handle(&handle);
     db_set_handle(&handle, fi->database, NULL);
-    if (db_open_database(driver, &handle) != DB_OK)
+    if (db_open_database(driver, &handle) != DB_OK) {
+        db_shutdown_driver(driver);
+        Vect_close(Map);
         G_fatal_error(_("Unable to open database <%s> by driver <%s>"),
                       fi->database, fi->driver);
+    }
     db_init_string(&table_name);
     db_set_string(&table_name, fi->table);
-    if (db_describe_table(driver, &table_name, &table) != DB_OK)
+    if (db_describe_table(driver, &table_name, &table) != DB_OK) {
+        db_close_database(driver);
+        db_shutdown_driver(driver);
```

Comment:
This can be simplified by  [db_close_database_shutdown_driver()](https://grass.osgeo.org/programming8/defs_2dbmi_8h.html#a671c7691d0fc81620a1831b6a5684360).